framework/src/headers/ip/v6/mod.rs

```diff
@@ -302,6 +310,31 @@ impl Ipv6Header {
     pub fn set_next_header(&mut self, hdr: NextHeader) {
         self.next_header = hdr as u8
     }
+
+    // Peek into the next header and a) if it's an SRH (Routing Header), then b)
+    // return the last_entry field +1 to give us the number of segments without
+    // parsing the SRH first.
+    #[inline]
+    pub fn peek_num_segments(&self) -> Option<u8> {
```

Comment:
Question for everyone: Does this make sense here or should it be something outside the Ipv6 header where you read the next header from the packet, read the field directly, and then reset to the previous spot?

@clinedome @pauljamescleary @drunkirishcoder @awangc

Reply:
@seancribbs i'm ok both to cut this or to make it a trait where each ext headr or v6, of course, impl it to peek ahead of the it sees a next header routing header. LMK.

Reply:
I have several concerns. First, this functionality cross-cuts the definition of any single header (looking into and interpreting another from outside), and cuts across abstraction boundaries (assumptions about the `Packet`/ `MBuf` layer). Second, it uses some unsafe operations where they shouldn't be necessary; if you're using `Packet` to accomplish the same goal the unsafe code is concentrated in `Packet` and the code around it is safe. Third, this is very application-specific functionality and not general enough across the possible extension headers that may follow the `Ipv6Header`; let's keep generic things in the framework and put application things outside.

Reply:
ok. @seancribbs cutting it for greater good.